objectivec/ort_value.mm

```diff
@@ -127,6 +155,29 @@ - (nullable NSMutableData*)tensorDataWithError:(NSError**)error {
   ORT_OBJC_API_IMPL_CATCH_RETURNING_NULLABLE(error)
 }
 
+- (nullable NSArray<NSString*>*)tensorStringDataWithError:(NSError**)error {
+  try {
+    const auto tensorTypeAndShapeInfo = _typeInfo->GetTensorTypeAndShapeInfo();
+    const size_t elementCount = tensorTypeAndShapeInfo.GetElementCount();
+    const size_t tensorStringDataLength = _value->GetStringTensorDataLength();
+    std::string tensorStringData(tensorStringDataLength, '\0');
+    std::vector<size_t> offsets(elementCount);
+    _value->GetStringTensorContent(const_cast<char*>(tensorStringData.data()), tensorStringDataLength,
+                                   offsets.data(), offsets.size());
+
+    NSMutableArray<NSString*>* result = [NSMutableArray arrayWithCapacity:elementCount];
+    for (size_t idx = 0; idx < elementCount; ++idx) {
+      const size_t strLength = (idx == elementCount - 1) ? tensorStringDataLength - offsets[idx]
+                                                         : offsets[idx + 1] - offsets[idx];
+      result[idx] = [[NSString alloc] initWithBytes:tensorStringData.data() + offsets[idx]
```

Comment:
should we use `addObject:`? I didn't know that this would work
https://developer.apple.com/documentation/foundation/nsmutablearray/1411274-addobject